Sources/NIO/Selector.swift

```diff
@@ -352,9 +355,17 @@ final class Selector<R: Registration> {
         case .now:
             ready = Int(try Epoll.epoll_wait(epfd: self.fd, events: events, maxevents: Int32(eventsCapacity), timeout: 0))
         case .blockUntilTimeout(let timeAmount):
-            var ts = itimerspec()
-            ts.it_value = timespec(timeAmount: timeAmount)
-            try TimerFd.timerfd_settime(fd: timerfd, flags: 0, newValue: &ts, oldValue: nil)
+            // Only call timerfd_settime if we not already scheduled one that will cover it.
+            // This guards against calling timerfd_settime if not needed as this is generally speaking
+            // expensive.
+            let next = DispatchTime.now().uptimeNanoseconds + UInt64(timeAmount.nanoseconds)
+            if next < earliestTimer {
+                earliestTimer = next
```

Comment:
Actually, please use explicit `self's on `earliestTimer`. It'll help make this a lot clearer.

Reply:
done